docs/docs/health-checks.md

```diff
@@ -16,9 +16,123 @@ application's tasks.
 A health check is considered passing if (1) its HTTP response code is between
 200 and 399 inclusive, and (2) its response is received within the
 `timeoutSeconds` period. If a task fails more than `maxConsecutiveFailures`
-health checks consecutively, that task is killed.
+health checks consecutively, that [task is terminated](#task-termination).
 
-##### Example usage
+## Mesos-level health checks versus Marathon-level health checks
+
+Marathon offers the following protocols for health checks. Declaring a protocol is optional: the default is `HTTP`.
+
+- `HTTP`
+- `HTTPS`
+- `TCP`
+- `COMMAND`
+- `MESOS_HTTP`
+- `MESOS_HTTPS`
+- `MESOS_TCP`
+
+### Marathon-level health checks
+
+Marathon-level health checks (HTTP, HTTPS, and TCP) are executed by Marathon and thus test reachability from the current Marathon leader. Marathon-level health checks have several limitations:
+
+- The API for health checks varies from one framework to the next and common functionality must be reimplemented with each new framework that requires health checks.
```

Comment:
This sounds like potentially a comment for framework developers and not users of Marathon. Does that sound right to you?